### PR TITLE
plugins: Enable linux-pipewire on FreeBSD

### LINKED ISSUE #4951: PipeWire capture fails to compile on FreeBSD
### Operating System Info

Other

### Other OS

FreeBSD

### OBS Studio Version

27.0.1

### OBS Studio Version (Other)

13

### OBS Studio Log URL

n/a

### OBS Studio Crash Log URL

_No response_

### Expected Behavior

 ```linux-capture``` should compile as before.

### Current Behavior

The ```linux-capture``` plugin worked on FreeBSD in 26.1.2 but in 27.0.1 it doesn't compile any more due to this linux-specific line: https://github.com/obsproject/obs-studio/blob/master/plugins/linux-capture/pipewire.c#L30

Users complain: https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=256862

### Steps to Reproduce

1.

### Anything else we should know?

_No response_

### PR DESCRIPTION
**Pull request by jbeich**
### Description
Depends on #7989

PipeWire capture requires few changes to support on FreeBSD >= 13.

### Motivation and Context
Wayland support under KDE/Plasma and GNOME ([not yet](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=258042)). Optional under wlroots as it can use [wlrobs](https://hg.sr.ht/~scoopta/wlrobs) instead.

Fixes #4951

### How Has This Been Tested?
```
# pkg install obs-studio sway pam_xdg xdg-desktop-portal-wlr xdg-desktop-portal
# echo 'session optional pam_xdg.so notroot runtime' >>/etc/pam.d/system
$ exit # log out to pick up XDG_RUNTIME_DIR
<new login>
$ dbus-daemon --session --fork --address=unix:runtime=yes
$ seatd-launch sway
<GUI session>
$ export XDG_CURRENT_DESKTOP=sway
$ dbus-update-activation-environment --all
$ daemon -f pipewire
$ daemon -f /usr/local/libexec/xdg-desktop-portal # workaround
$ daemon -f /usr/local/libexec/xdg-desktop-portal-wlr # workaround
$ obs
[...]
info: [pipewire] Available captures:
info: [pipewire]     - Desktop capture
[...]

<Sources>
<Add Source>
<Screen Capture (PipeWire)
<OK>
[...]
info: [pipewire] Screencast session created
info: [pipewire] Asking for desktop
info: [pipewire] desktop selected, setting up screencast
error: Cannot query the number of formats: An EGLDisplay argument does not name a valid EGL display connection.
info: [pipewire] Server version: 0.3.62
info: [pipewire] Library version: 0.3.62
info: [pipewire] Header version: 0.3.62
info: [pipewire] Created stream 0x8c5f97c00
info: [pipewire] Stream 0x8c5f97c00 state: "connecting" (error: none)
info: [pipewire] Playing stream 0x8c5f97c00
info: [pipewire] Stream 0x8c5f97c00 state: "paused" (error: none)
info: [pipewire] Negotiated format:
info: [pipewire]     Format: 7 (Spa:Enum:VideoFormat:RGBx)
info: [pipewire]     Size: 3840x2160
info: [pipewire]     Framerate: 0/1
info: [pipewire] Stream 0x8c5f97c00 state: "streaming" (error: none)
[...]

<Start Recording>
<Stop Recording>
```

### Types of changes
- New feature (non-breaking change which adds functionality)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by tytan652** [comment]
Put this PR as draft if it depends on another one, the link to do that is inside the column on the right of the page.

**Comment by jbeich** [comment]
Given FreeBSD CI is already failing (without any of my changes) there's no point to block review/merge on another PR by marking as a draft. Instead this was tested (and backported) downstream via poudriere on FreeBSD 13.1 amd64/i386, 14.0 amd64.

**Comment by Fenrirthviti** [comment]
Regardless of the CI status, drafts that depend on other PRs to be merged first should be marked as drafts until the dependency is merged. It's to prevent accidental merging out of order. It shouldn't impact any kind of review process.

**Review by kkartaltepe — APPROVED** [review]
This looks fine to me

**Comment by jbeich** [comment]
The dependency has been merged, so this one can as well.

File: CI/freebsd/01_install_dependencies.sh [hunk 1]
```diff
@@ -24,7 +24,11 @@ install_obs-deps() {
     status "Install OBS dependencies"
     trap "caught_error 'install_obs-deps'" ERR
 
-    sudo pkg install -U -y $@
+    if [ -z "${DISABLE_PIPEWIRE}" ]; then
+	sudo pkg install -U -y $@ pipewire
+    else
+	sudo pkg install -U -y $@
+    fi
```
Comment (RytoEX) [inline CI/freebsd/01_install_dependencies.sh:31]: When running this script by itself, it is impossible to set `DISABLE_PIPEWIRE` despite the usage message. You need to allow this script to set this value. See the `linux/01_install_dependencies.sh` scripts for an example.

**Review by RytoEX — CHANGES_REQUESTED** [review]

File: CI/freebsd/01_install_dependencies.sh (hunk 1 above)
Comment (jbeich) [reply CI/freebsd/01_install_dependencies.sh:31]: Somehow forgot to copy that from Linux config. Done.

**Comment by jbeich** [comment]
#4951 was closed before this was merged. Are FreeBSD fixes no longer "welcome"?

**Comment by kkartaltepe** [comment]
> #4951 was closed before this was merged. Are FreeBSD fixes no longer "welcome"?

We generally close issues very aggressively, I have re-opened it if you prefer it to be closed with this PR.

**Comment by RytoEX** [comment]
> #4951 was closed before this was merged. Are FreeBSD fixes no longer "welcome"?

We just merged FreeBSD fixes, from you, in fact, within the past few weeks, and then we released those fixes as part of OBS Studio 29.0.1+. The project's stance on FreeBSD support has not changed. We continue to rely on the FreeBSD developer community to submit and test PRs as needed, as none of the regular contributors runs FreeBSD as a daily driver.

As @kkartaltepe stated above, we tend to aggressively close Issues, and that one just got caught in an action item that said, loosely, "close or merge linked PR". It has been reopened.

**Review by RytoEX — CHANGES_REQUESTED** [review]
Please change the first commit's prefix to just "linux-pipewire: ". I would prefer that all the CI commits just began with "CI: ". The only sub-folder we give a specific prefix is "flatpak".

**Comment by jbeich** [comment]
Ah, I could just force push earlier state then push the one with fixed commit messages. Now one can see/confirm nothing functional has changed since the last review by clicking on the last "force-pushed" link.

File: CI/freebsd/01_install_dependencies.sh [hunk 2]
```diff
@@ -79,6 +83,7 @@ print_usage() {
             "-h, --help                     : Print this help\n" \
             "-q, --quiet                    : Suppress most build process output\n" \
             "-v, --verbose                  : Enable more verbose build process output\n"
+            "--disable-pipewire             : Disable building with Pipewire support (default: off)\n"
```
Comment (RytoEX) [inline CI/freebsd/01_install_dependencies.sh:86]: Nit: Pipewire -> PipeWire

**Review by RytoEX — CHANGES_REQUESTED** [review]
Commit message nit:
* pipewire -> PipeWire

When writing PipeWire in descriptive text, capitalize it appropriately.

Other than nits, this should be good to land.

File: CI/freebsd/01_install_dependencies.sh (hunk 2 above)
Comment (jbeich) [reply CI/freebsd/01_install_dependencies.sh]: `Pipewire` spelling was inherited from Linux config, so I had to fix there as well.

**Comment by jbeich** [comment]
> When writing PipeWire in descriptive text, capitalize it appropriately.

The reference was ambiguous: `pipewire` as package or `PipeWire` as project.

**Pull request merged**